modules/gapi/src/executor/last_value.hpp

```diff
@@ -75,6 +87,12 @@ void last_written_value<T>::push(const T& t) {
 template<typename T>
 void last_written_value<T>::pop(T &t) {
     std::unique_lock<std::mutex> lock(m_mutex);
+    if (m_data.has_value() && m_drop_strategy.has_value()) {
+        auto& strategy = m_drop_strategy.value();
+        if (strategy(m_data.value())) {
+            m_data.reset();
+        }
+    }
```

Comment:
Comments would be welcome here.. Further generations would thank you for that :)

Reply:
Done